src/Uno.UI/UI/Xaml/Documents/Run.skia.cs

```diff
@@ -45,102 +49,144 @@ private List<Segment> GetSegments()
 				int lineBreakLength = 0;
 				bool wordBreakAfter = false;
 
-				// Count leading spaces
-
-				while (i < text.Length && char.IsWhiteSpace(text[i]) && !Unicode.IsLineBreak(text[i]))
+				if (symbolTypeface is { })
 				{
-					leadingSpaces++;
-					i++;
+					var fi = FontDetailsCache.GetFont(symbolTypeface.FamilyName, (float)FontSize, FontWeight, FontStyle);
+					font = fi.Font;
+					wordBreakAfter = Unicode.HasWordBreakOpportunityAfter(text, i) || (i + 1 < text.Length && Unicode.HasWordBreakOpportunityBefore(text, i + 1));
+					font.GetScale(out fontScale, out _);
+					textSizeY = fontSize / fontScale;
+					textSizeX = textSizeY * paint.TextScaleX;
+					i += surrogate ? 2 : 1;
+					surrogate = false;
+					fallbackFont = fi;
+					symbolTypeface = default;
 				}
-
-				// Keep the segment going until we hit a word break opportunity or a line break
-
-				while (i < text.Length)
+				else
 				{
-					if (ProcessLineBreak(text, ref i, ref lineBreakLength))
-					{
-						break;
-					}
+					// Count leading spaces
 
-					if (Unicode.HasWordBreakOpportunityAfter(text, i) || (i + 1 < text.Length && Unicode.HasWordBreakOpportunityBefore(text, i + 1)))
+					while (i < text.Length && char.IsWhiteSpace(text[i]) && !Unicode.IsLineBreak(text[i]))
 					{
-						if (char.IsWhiteSpace(text[i]))
-						{
-							trailingSpaces++;
-						}
-
-						wordBreakAfter = true;
+						leadingSpaces++;
 						i++;
-						break;
 					}
 
-					i++;
-				}
+					// Keep the segment going until we hit a word break opportunity or a line break
 
-				// Tack on any trailing spaces or line breaks if this segment does not yet end in a line break
-
-				if (lineBreakLength == 0)
-				{
 					while (i < text.Length)
 					{
 						if (ProcessLineBreak(text, ref i, ref lineBreakLength))
 						{
 							break;
 						}
 
-						if (char.IsWhiteSpace(text[i]))
+						if (Unicode.HasWordBreakOpportunityAfter(text, i) || (i + 1 < text.Length && Unicode.HasWordBreakOpportunityBefore(text, i + 1)))
 						{
-							trailingSpaces++;
+							if (char.IsWhiteSpace(text[i]))
+							{
+								trailingSpaces++;
+							}
+							wordBreakAfter = true;
 							i++;
+							break;
+						}
+
+						if (char.IsSymbol(text[i]))
+						{
+							symbolTypeface = SKFontManager.Default
+								.MatchCharacter(text[i]);
+							break;
 						}
-						else
+						else if (i + 1 < text.Length
+							&& char.IsSurrogate(text[i])
+							&& char.IsSurrogatePair(text[i], text[i + 1]))
 						{
+							var lang = LanguageTagCache.GetLanguageTag(this);
+							var fontManager = SKFontManager.Default;
+							var codepoint = (int)((text[i] - 0xD800) * 0x400 + (text[i + 1] - 0xDC00) + 0x10000);
+							symbolTypeface = fontManager
+								.MatchCharacter(null, SKFontStyle.Normal, lang, codepoint);
```

Comment:
It's somewhat confusing that culture/lang should affect the rendering of anything.